src/Measurements.jl

```diff
@@ -77,16 +81,21 @@ function Measurement{T}(::S) where {T, S}
 end
 
 # Functions to quickly create an empty Derivatives object.
-@generated empty_der1(x::Measurement{T}) where {T<:AbstractFloat} = Derivatives{T}()
-@generated empty_der2(x::T) where {T<:AbstractFloat} = Derivatives{x}()
+@generated empty_der1(x::Measurement{T}) where {T<:Real} = Derivatives{T}()
+@generated empty_der2(x::T) where {T<:Real} = Derivatives{x}()
 
 # Start from 1, 0 is reserved to derived quantities
 const tag_counter = Threads.Atomic{UInt64}(1)
 
 measurement(x::Measurement) = x
-measurement(val::T) where {T<:AbstractFloat} = Measurement(val, zero(T), UInt64(0), empty_der2(val))
+
+function _measurement(val::T) where {T<:Real}
+    return Measurement(val, zero(T), UInt64(0), empty_der2(val))
+end
```

Comment:
```suggestion
_measurement(val::T) where {T<:AbstractFloat} = Measurement(val, zero(T), UInt64(0), empty_der2(val))
```
would minimize the diff or am I missing something?

Reply:
Note the discrepancy in `{T<:AbstractFloat}` vs `{T<:Real}` in suggested diff, i'm guessing that is unintentional?